### ISSUE TITLE
Address Book implementation

### ISSUE THREAD

**Issue by holmholk** [turtlecoin/mobile-kotlin#8]
Currently Address Book screen is not fully implemented. We need to implement:

**Add Contacts**
Add and store contacts locally

**Show Contact List**
Get a list of contacts from the local storage and display it to Address Book

**View Contacts**
Display the contact's address and QR code

**Edit Contacts**
Pre-populate contacts data and fill each field. Update the storage to save.

**Delete Contacts**
Delete contacts from the local storage and update the Address Book list afterwards

**Search Contacts**
Search by name function using SearchView. Search box should be in the action bar, or below it if that's easier.

(future feature)
**Restore Contacts**
We want users to be able to export and import contacts to restore data when they want to use different devices

Feel free to talk more about the details

**Comment by sslp** [comment]
I'm looking into [Room Library](https://developer.android.com/training/data-storage/room/index.html) right now, for saving User <-> Address into a local DB

Looking [here](https://developer.android.com/training/data-storage/index.html), we we best fits are Room or normal SQLite.

**Comment by holmholk** [comment]
Yeah it looks perfect for our use case. Room is probably the way to go

**Comment by sslp** [comment]
I can successfully use the DB now. But there's a lot of features to add 8)
